Typography CSS Changes

woodrowbarlow opened the pull request:
this addresses issues #3 #10 and #17.

it keeps your `style.css` (with some very minor changes) and adds a `typography.css`.

Commit:
template changes and first partial styling draft

this is the basic styling for viewing an entry. styling for other
pages (such as editing an entry) is not yet done.

there are some things in here that jdpage doesn't agree with from
a code philosophy standpoint. i'm leaving them in until i have a
full working prototype and then i'll be considering compromises at
a later date.

changes of note:

 * importing jason pamental's font scale.
 * replace normalize with reset, since all font sizes are explicitly
   defined now.
 * templates move the body tag out of main and into sub-templates,
   so that classes can be applied.
 * mathjax is now only loaded on the entry page, and loaded at the
   end of the document to speed up download times.
 * pages which have a nav now have the same nav duplicated in the
   header.

Commit:
separate typography from visual style

the typography stylesheet simply defines font sizes, families,
and spacings.

Commit:
use (basically) the existing style

rather than the stylesheet i'm working on, this commit uses the
stylesheet that is currently live. some small changes were made to
make it play well with the external typography file.

Commit:
remove iconfont

it isn't needed for the typography PR.

jdpage commented:
Okay, up front: visited internal links and visited external links look exactly the same. This needs to be fixed before I merge this, either with a colour change or some other indicator. Also, you removed the MathJax script from the templates, which is a regression.

Other than that, I like the new link colours! The new blue/red/green for unvisited links are very pleasant to look at. The underline on the `h2` elements is a nice touch. Justifying works surprisingly well; I usually run into trouble with it, due to the lack of browser hyphenation. Consider adding a hyphenator script? I note that you have not justified list elements. Is this an oversight?

I have reservations about the font choices. Google Fonts strips out all of the hinting information on their fonts, which means that (on my machine, where I don't have Ubuntu font installed locally), the kerning goes to hell at small sizes. We might be better off sticking with the classic Helvetica-Arial/Georgia/anything-but-Courier-New (I know, it's boring, but it's also really readable).

There seems to be a vertical spacing issue on the links at the bottom of the page. Also, I miss the hanging bullets (where they stick out into the margin), but if you really don't like them I guess they can go.

jdpage commented:
Can I suggest the progression `'Lucida Grande', 'Lucida Sans Unicode', 'Helvetica', 'Arial', sans-serif` for the body? The Lucida fonts are quite handsome, eminently readable, and Grande comes installed on OS X while Sans Unicode comes installed on Windows.

jdpage commented:
... also, my suggestion of Georgia for the titles was entirely regrettable. `"Palatino Linotype", "Book Antiqua", Palatino, serif`, perhaps?

Commit:
first batch of fixes

this addresses the following:

 * mathjax had been accidentally ommitted, has been added back
 * contrast between dark blue and dark green was too low
 * list elements were left-aligned, now they are justified
 * list styling was not applied to the nav at the bottom

this does not yet address the fonts.

woodrowbarlow commented:
here's a sample of the new colors

![color sample](http://i.imgur.com/lGUbQr0.png)

jdpage commented:
I like it. Since "broken" links are links to pages that don't exist yet, they can't be visited (you can navigate to the URL, but can't actually "visit" it, technically), so what do you think about making `a.broken` and `a.broken:visited` the same colour?

woodrowbarlow commented:
yeah. that's actually how i had it originally, for the same reason. not really sure why i changed it, so i'll change it back.

Commit:
make broken links look the same even when they're visited

jdpage commented:
Bug. Because you have the little link-sigils set to `display: inline-block`, this happens:

![spacing nooooooo](http://i.imgur.com/NKF7epY.png)

If you set it to `display: inline`, you get this spacing (but that irritating little underline)

![spacing ok](http://i.imgur.com/qgMkdSh.png)

If you used a non-breaking space inside the content instead of that padding/margin thing, then you wouldn't get that break in the underline, though.

woodrowbarlow commented:
i think i've settled on papyrus for the body and comic sans for headers. i'll whip up a demo for you later. also ryan is over, do you know what time you will be home?

Commit:
formatting

Commit:
fix link sigil spacing problem

Commit:
add lang attribute to templates

including language information allows the browser to wrap text more
intelligently. this commit sets language on all pages to english.

in the future, this may be something that should be set by a
template variable stored in the .conf file.

woodrowbarlow commented:
**text wrapping**: most modern browsers now at least partially support the CSS [hyphens](https://developer.mozilla.org/en-US/docs/Web/CSS/hyphens) property, which tells the browser that it may hyphenate words when wrapping wherever it deems appropriate. i've implemented that property now. i am still considering looking for supplemental scripts that would provide the same usability in browsers that haven't implemented language dictionaries yet.

**bullet indents**: in order to have bullets hanging negatively into the margin, the body of the list item would need to be flush with the content. but since paragraphs are indented, it wouldn't make much sense to have text from bullet points flush. i gave them the same indentation that the paragraphs have.

**font kerning**: can you post you exact browser / OS version and a few screenshots of the kerning issues you're seeing? i am temporarily removing the web fonts, just so this PR can be resolved... but i feel nervous doing so. i'm nervous that you won't even consider adding them back in once you get a PR that has them removed.

**visited broken links**: are now the same color as normal broken links.

**link sigil spacing**: is now fixed. the text-indent property was cascading down inappropriately.

Commit:
temporarily remove web fonts

for the sake of resolving the pull request, i am temporarily taking
out the web fonts. i still think they could work.

Commit:
add sigils for email and doi links
